File: BackEnd/backend/src/main/java/com/fitapp/backend/feedback/aplication/dto/request/CreateFeedbackRequest.java
package com.fitapp.backend.feedback.aplication.dto.request;

import com.fitapp.backend.feedback.domain.model.FeedbackCategory;
import com.fitapp.backend.feedback.domain.model.FeedbackType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.Map;

@Data
public class CreateFeedbackRequest {

    @NotNull(message = "type is required")
    private FeedbackType type;

    private FeedbackCategory category;

    @Size(max = 200, message = "title must be at most 200 characters")
    private String title;

    @NotBlank(message = "message is required")
    @Size(min = 10, message = "message must be at least 10 characters")
    private String message;

    private String stepsToReproduce;

    private Boolean includeTechnicalContext;

    private Map<String, String> technicalContext;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/feedback/model/request/CreateFeedbackRequest.kt
package com.fitapp.appfit.feature.feedback.model.request

import com.google.gson.annotations.SerializedName

data class CreateFeedbackRequest(
    @SerializedName("type") val type: String,
    @SerializedName("category") val category: String?,
    @SerializedName("title") val title: String?,
    @SerializedName("message") val message: String,
    @SerializedName("stepsToReproduce") val stepsToReproduce: String?,
    @SerializedName("includeTechnicalContext") val includeTechnicalContext: Boolean,
    @SerializedName("technicalContext") val technicalContext: Map<String, String>?
)
